src/App.tsx

```diff
@@ -1,34 +1,62 @@
 import { useKeycloak } from "@react-keycloak/web";
-import { Route, Routes, Navigate } from 'react-router-dom';
+import { Route, Routes, Navigate } from "react-router-dom";
 import { useEffect, useState } from "react";
-import { api } from './api/api';
-import Home from './pages/Home';
-import CaretakerForm from './pages/CaretakerForm';
-import LoginPage from './pages/LoginPage';
-import store from './store/RootStore';
+import { api } from "./api/api";
+import Home from "./pages/Home";
+import CaretakerForm from "./pages/CaretakerForm";
+import LoginPage from "./pages/LoginPage";
+import store from "./store/RootStore";
 import CaretakerSearch from "./pages/CaretakerSearch";
+import ProfileSelection from "./pages/ProfileSelection";
 
 function App() {
   const { keycloak, initialized } = useKeycloak();
+  const [isXsrfTokenFetched, setIsXsrfTokenFetched] = useState(false);
+  const [isUserDataFetched, setIsUserDataFetched] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     const fetchXsrfToken = async () => {
       if (keycloak.authenticated && !store.user.xsrfToken) {
         await api.getXsrfToken();
+        setIsXsrfTokenFetched(true);
+      }
+    };
+
+    const fetchUserData = async () => {
+      try {
+        const userData = await keycloak.loadUserProfile();
+        const userProfileData = {
+          username: userData.username,
+          email: userData.email,
+          firstName: userData.firstName,
+          lastName: userData.lastName,
+          token: store.user.xsrfToken,
+          selected_profile: null,
+        };
+        store.user.saveProfileToStorage(userProfileData);
+        setIsUserDataFetched(true);
+      } catch (error) {
+        console.error(`Failed to load user profile: ${error}`);
       }
-      setIsLoading(false);
     };
 
     if (initialized) {
       if (keycloak.authenticated) {
         fetchXsrfToken();
+        fetchUserData();
       } else {
         setIsLoading(false);
       }
     }
   }, [initialized, keycloak.authenticated]);
```

Comment:
Tutaj `eslint` woła, że brakuje mu zależności `keycloak` w tablicy zależności. Niby ma racje, ale tymczasowo dodałbym linijkę, żeby ignorowało to: 
```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [initialized, keycloak.authenticated]);
```
A docelowo trzebaby to przemyśleć i poprawić (pewnie przy użyciu `useCallback`).
Ja w tych zależnościach przy używaniu `useEffect` też sam się gubię, a nie chce robić tego na wyczucie więc na razie niech będzie ignorowane przez `eslinta`.